File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/viewmodels/OnboardingViewModel.kt
package family.aladdin.android.viewmodels

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

/**
 * 👋 Onboarding View Model
 * Логика для экрана онбординга
 * Источник: iOS OnboardingViewModel.swift
 */

class OnboardingViewModel : ViewModel() {
    
    // MARK: - State
    
    private val _currentPage = MutableStateFlow(0)
    val currentPage: StateFlow<Int> = _currentPage
    
    private val _isCompleted = MutableStateFlow(false)
    val isCompleted: StateFlow<Boolean> = _isCompleted
    
    val totalPages: Int = 4
    
    // MARK: - Public Methods
    
    /**
     * Перейти на следующую страницу
     */
    fun nextPage() {
        if (_currentPage.value < totalPages - 1) {
            _currentPage.value += 1
        } else {
            completeOnboarding()
        }
    }
    
    /**
     * Перейти на предыдущую страницу
     */
    fun previousPage() {
        if (_currentPage.value > 0) {
            _currentPage.value -= 1
        }
    }
    
    /**
     * Пропустить онбординг
     */
    fun skipOnboarding() {
        completeOnboarding()
    }
    
    /**
     * Перейти на конкретную страницу
     */
    fun goToPage(page: Int) {
        if (page in 0 until totalPages) {
            _currentPage.value = page
        }
    }
    
    // MARK: - Private Methods
    
    /**
     * Завершить онбординг
     */
    private fun completeOnboarding() {
        _isCompleted.value = true
        
        // В production: сохранить в SharedPreferences
        // SharedPreferences.edit().putBoolean("hasCompletedOnboarding", true).apply()
        
        println("Onboarding completed, navigating to main screen")
    }
    
    /**
     * Сбросить онбординг (для тестирования)
     */
    fun resetOnboarding() {
        _currentPage.value = 0
        _isCompleted.value = false
    }
}

File: BACKUPS/BACKUP_MOBILE_20251128_164948/ViewModels/OnboardingViewModel.swift
import SwiftUI
import Security

/// 👋 Onboarding View Model
/// Логика для экрана онбординга
/// ⚠️ ВНИМАНИЕ: Этот ViewModel НЕ используется в 14_OnboardingScreen.swift
/// Он используется только в других версиях онбординга (если есть)
/// Текущий онбординг использует свой массив pages с 7 страницами!
class OnboardingViewModel: ObservableObject {
    
    @Published var currentPage: Int = 0
    @Published var isCompleted: Bool = false
    
    // ⚠️ УСТАРЕВШЕЕ: Это значение не используется в 14_OnboardingScreen.swift
    // Реальный онбординг использует pages.count (должно быть 7)
    // ❌ НЕ ИСПОЛЬЗУЕТСЯ в текущем онбординге!
    let totalPages: Int = 4
    
    func nextPage() {
        if currentPage < totalPages - 1 {
            currentPage += 1
        } else {
            completeOnboarding()
        }
    }
    
    func skipOnboarding() {
        completeOnboarding()
    }
    
    private func completeOnboarding() {
        isCompleted = true
        // Сохраняем статус онбординга
        // TODO: В будущем заменить на Keychain для безопасности
        UserDefaults.standard.set(true, forKey: AppConfig.UserDefaultsKeys.hasCompletedOnboarding)
    }
}



